aten/src/ATen/native/cuda/UnarySignKernels.cu

```diff
@@ -74,9 +74,18 @@ void sign_kernel_cuda(TensorIteratorBase& iter){
 }
 
 void signbit_kernel_cuda(TensorIteratorBase& iter){
-  AT_DISPATCH_ALL_TYPES_AND2(kBFloat16, ScalarType::Half, iter.input_dtype(), "signbit_cuda", [&]() {
-    gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> bool { return is_negative(a); });
-  });
+  // NOTE: signbit does not always support integral arguments.
+  if (at::isIntegralType(iter.input_dtype(), /*includeBool=*/false)) {
```

Comment:
Hmm, perhaps I'm reading this code wrong, signbit will not throw exception for booleans, isn't it? (as its not included in integral type, and its clearly not a float?)

Reply:
This is a little confusing without context, but the signbit operation has a short-circuit for bool, so there's no kernel for it:

https://github.com/pytorch/pytorch/blob/059c439ed9bd675102a0c6cf1fcf907c88288960/aten/src/ATen/native/UnaryOps.cpp#L208